tests/test_olio_grid_functions.py

```diff
@@ -0,0 +1,306 @@
+    ((0, 1), (1, 2), np.array([[True, False], [True, True]], dtype = bool)),
+    ((2, 1), (1, 0), np.array([[True, True], [False, True]], dtype = bool)),
+    ((2, 1), (0, 1), np.array([[False, True], [False, True]], dtype = bool)),
+    ((2, 1), (1, 2), np.array([[False, False], [False, True]], dtype = bool)),
+])
+def test_left_right_foursome(previous, next, expected_result):
+
+    # --------- Arrange ----------
+    full_p_list = [previous, (1, 1), next]
+    p_index = 1
+    # --------- Act ----------
+    result = left_right_foursome(full_pillar_list = full_p_list, p_index = p_index)
+    # --------- Assert ----------
+    np.testing.assert_equal(result, expected_result)
+
+
+def test_left_right_foursome_error_handling():
+
+    # --------- Arrange ----------
+    p_index = 1
+    # arrange for exceptions
+    full_p_list_entry_error = [(1, 0), (1, 2), (0, 1)]
+    full_p_list_exit_error = [(1, 0), (1, 1), (2, 2)]
+
+    # --------- Act and Assert----------
+    with pytest.raises(Exception) as excinfo:
+        result_entry_error = left_right_foursome(full_pillar_list = full_p_list_entry_error, p_index = p_index)
+        assert "code failure whilst taking entry sides from dubious full pillar list" in str(excinfo.value)
+    with pytest.raises(Exception) as excinfo2:
+        result_exit_error = left_right_foursome(full_pillar_list = full_p_list_exit_error, p_index = p_index)
+        assert "code failure whilst taking exit sides from dubious full pillar list" in str(excinfo.value)
+
+
+def test_resequence_nexus_corp():
+
+    # --------- Arrange ----------
+    corner_points = np.array([[[[[[[0., 0., 0.], [10., 0., 0.]], [[10., -10., 0.], [0., -10., 0.]]],
+                                 [[[0., 0., 10.], [10., 0., 10.]], [[10., -10., 10.], [0., -10., 10.]]]]]]])
+
+    expected_result = np.array([[[[[[[0., 0., 0.], [10., 0., 0.]], [[0., -10., 0.], [10., -10., 0.]]],
+                                   [[[0., 0., 10.], [10., 0., 10.]], [[0., -10., 10.], [10., -10., 10.]]]]]]])
+
+    # --------- Act ----------
+    resequence_nexus_corp(corner_points = corner_points)
+
+    # --------- Assert ----------
+    np.testing.assert_almost_equal(corner_points, expected_result)
+
+
+def test_random_cell(tmp_path):
+
+    # --------- Arrange----------
+    epc = os.path.join(tmp_path, 'grid.epc')
+    model = rq.new_model(epc)
+
+    # create a basic block grid
+    dxyz = (10.0, 10.0, 100.0)
+    grid = grr.RegularGrid(model,
+                           extent_kji = (3, 2, 2),
+                           title = 'grid_1',
+                           origin = (0.0, 10.0, 1000.0),
+                           dxyz = dxyz,
+                           as_irregular_grid = True)
+
+    grid_points = grid.points_ref(masked = False)
+
+    # pinch out cells in the k == 1 layer
+    for j in range(3):
+        for i in range(3):
+            grid_points[2][j][i][-1] = 1100.
+    grid.write_hdf5()
```

Comment:
The write_hdf5(), create_xml(), store_epc() should happen later, after all modifications of grid_points